lib/Scheduler/Scheduler.cpp

```diff
@@ -22,19 +22,19 @@ bool isTTIROp(mlir::Operation *op) {
 }
 
 bool isTTShedulableOp(mlir::Operation *op) {
-  return isTTNNOp(op) || isTTIROp(op);
+  return (isTTNNOp(op) || isTTIROp(op)) && (not isa<func::ReturnOp>(op));
```

Comment:
![image](https://github.com/user-attachments/assets/7f16b60b-a886-49cb-8bdc-f98209c0a4c7)

I think it would be better to check traits and see if its "return-like".